### ISSUE TITLE
Crash on export of settings (for K-9 Mail in managed profile)

### ISSUE THREAD

**Issue by engelhro** [thunderbird/thunderbird-android#3659]
# Environment

* **Device**: HTC One M8
* **OS**: LineageOS 14.1 / Android 7.1.2 (rooted)
* **K-9 Mail**: version 5.600

# Problem description

When trying to export my settings, K-9 Mail immediately crashes (FC).

This *might* be exactly the same issue as described in #3614, but there's one twist: I'm using K-9 Mail in a **managed profile** (i.e., "Android for Work").

As I'm not sure whether this plays a role and makes the described crash a different problem or not, I decided to open a new issue. Please note that the export for a managed profile should happen within the path `/storage/emulated/10/…` instead of the usual `/storage/emulated/0/…` (export path cannot be defined or checked anywhere in K-9 Mail?)

## Log

A debug log as described (here)[https://github.com/k9mail/k-9/wiki/LoggingErrors] is attached.

[Debuglog K9-Mail.txt](https://github.com/k9mail/k-9/files/2472694/Debuglog.K9-Mail.txt)

**Comment by cketti** [comment]
```
android.content.ActivityNotFoundException: No Activity found to handle Intent {
    act=android.intent.action.CREATE_DOCUMENT 
    cat=[android.intent.category.OPENABLE] 
    typ=application/octet-stream (has extras) 
}
```

It crashes when we try to display the (system) screen where the user can select where to create the settings file. Is it possible to completely prevent apps in the work profile from accessing storage?

**Comment by engelhro** [comment]
That might be the case, but it's not set like that. I can both access storage from other apps as well as K-9 Mail itself (it's possible to save attachments).

Also **import** of a configuration file works fine – in this case the selection screen for a file manager comes up, and I can then navigate to a file. But when trying to export, the crash happens *immediately*, no selection screen for the respective file manager appears…

**Comment by markd89** [comment]
I also get a crash on export and am not using Android for Work.
I'm on Lineage 15.1

**Comment by engelhro** [comment]
> I hope that helps.

I wouldn't count on it… Just look at sheer number of issues and open bug reports and the rate of fixes. Apparently k-9 is in, let's say, "low maintenance mode".

I meanwhile switched my smartphone, so: different device, different OS (and different Android version), non-rooted this time, and still the same problem. k-9 is simply not designed nor tested for managed profiles. I guess we have to live with that. No export, no backup of settings. 😩

**Comment by dschrempf** [comment]
I am having the same issure without using a managed profile. However, I am using Android 10.
The log is:

AndroidRuntime: Shutting down VM
01-13 11:29:35.447  8586  8586 E AndroidRuntime: FATAL EXCEPTION: main
01-13 11:29:35.447  8586  8586 E AndroidRuntime: Process: com.fsck.k9, PID: 8586
01-13 11:29:35.447  8586  8586 E AndroidRuntime: android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.intent.action.CREATE_DOCUMENT cat=[android.intent.category.OPENABLE] typ=application/octet-stream (has extras) }
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at android.app.Instrumentation.checkStartActivityResult(Instrumentation.java:2058)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1716)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at android.app.Activity.startActivityForResult(Activity.java:5210)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at android.app.Activity.startActivityForResult(Activity.java:5168)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at com.fsck.k9.activity.Accounts.onExport(Accounts.java:1933)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at com.fsck.k9.activity.Accounts.onOptionsItemSelected(Accounts.java:1265)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at android.app.Activity.onMenuItemSelected(Activity.java:4155)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at com.android.internal.policy.PhoneWindow.onMenuItemSelected(PhoneWindow.java:1257)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at com.android.internal.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:787)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at com.android.internal.view.menu.SubMenuBuilder.dispatchMenuItemSelected(SubMenuBuilder.java:85)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at com.android.internal.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:151)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at com.android.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:934)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at com.android.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:924)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at com.android.internal.view.menu.MenuPopup.onItemClick(MenuPopup.java:128)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at android.widget.AdapterView.performItemClick(AdapterView.java:330)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at android.widget.AbsListView.performItemClick(AbsListView.java:1192)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at android.widget.AbsListView$PerformClick.run(AbsListView.java:3200)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at android.widget.AbsListView$3.run(AbsListView.java:4118)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:883)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:100)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:214)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7356)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
01-13 11:29:35.447  8586  8586 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)

**Comment by cketti** [comment]
@dschrempf: What device are you using? Are you using a custom ROM? This is a core Android feature that should be present.

We can certainly change our code to not crash. But that won't change the fact that export won't work on affected devices.

**Comment by dschrempf** [comment]
I am using a custom ROM. My device is an LG G6. If it is not possible to support more customized setups, maybe the error message can be improved so that I know why the crash is happening?

The export feature has been working on the same ROM and setup until Android 9!

Thank you!

**Comment by cketti** [comment]
This should be fixed in K-9 Mail 5.7xx.

**Comment by dschrempf** [comment]
It turns out that I didn't have the Google File manager installed (I don't remember the actual name of the application). It was not possible to choose the save location, that is why K9Mail crashed. I had other file managers installed, though, but obviously they did not provide the CREATE_DOCUMENT action. Thanks for fixing this!
